application-diagram-api/src/main/java/com/xwiki/diagram/internal/handlers/DiagramMacroSolrMetadataExtractor.java

```
@@ -65,32 +73,81 @@ public class DiagramMacroSolrMetadataExtractor implements SolrEntityMetadataExtr
+     * @return true if all the references are valid, false otherwise.
+     */
+    private boolean updateMacroReference(XWikiDocument document, XDOM xdom, List<Block> macroBlocks)
+    {
+        try {
 
+            XWikiContext context = contextProvider.get();
+            boolean modified = false;
+            for (Block macroBlock : macroBlocks) {
+                String referenceName =
+                    macroBlock.getParameter(REFERENCE) != null ? macroBlock.getParameter(REFERENCE) : "Diagram";
+                // For backwards compatibility we check if the page already exists so we won't modify it.
+                DocumentReference macroReference =
+                    explicitDocumentReferenceResolver.resolve(referenceName, document.getDocumentReference());
+                boolean diagramAlreadyExists = context.getWiki().exists(macroReference, context);
+                if (!diagramAlreadyExists) {
+                    // First we check if the name is valid in the current naming strategy.
+                    boolean isValid = this.isValid(referenceName);
+                    // If the name is valid then we can use it, otherwise we transform the name in a valid
+                    // one and update the macro block.
+                    if (!isValid) {
+                        String transformedName = this.transformName(referenceName);
+                        macroBlock.setParameter(REFERENCE, transformedName);
+                        modified = true;
+                    }
+                }
+            }
+            if (modified) {
+                document.setContent(xdom);
+                context.getWiki()
+                    .saveDocument(document, "Updated diagram macro references to respect the name strategy.", context);
+                return modified;
+            }
+        } catch (XWikiException e) {
+            logger.error(e.getMessage(), e);
+            throw new RuntimeException(e);
```

I'm not sure you need to throw an error. Maybe just `return false`?